### PR TITLE
Use "npm-run-all" to run scripts

### PR DESCRIPTION
**Pull request by danawoodman**
[npm-run-all](https://github.com/mysticatea/npm-run-all) is a cross-browser approach to running npm tasks in series (`-s`) or parallel (`-p`). Rather than using a platform specific method to run scripts (e.g. `&&`), npm-run-all will work on all platforms and supports using Yarn if running with Yarn. I made a guess as to whether certain scripts should be run in series or parallel but feel free to change if my choices aren't optimal.

### CODE REVIEW

**Comment by KyleAMathews** [comment]
Deploy preview for *using-glamor* failed.

Built with commit d6ba54fe3abac1fcee6cd0dc9965734c646e5a11

https://app.netlify.com/sites/using-glamor/deploys/5a720f03efbe5d52bb9722eb

**Comment by danawoodman** [comment]
Forgot to add `yarn.lock` one minute...

**Comment by danawoodman** [comment]
@KyleAMathews I'm not sure why this is failing on Travis. The `bootstrap` command seems to be working fine for me. Also, locally the test suite fails which is strange...

**Comment by jquense** [comment]
what is the advantage to using this over lerna, which is already there.

**Comment by danawoodman** [comment]
@jquense it solves a different problem which I describe in the description. Basically lerna runs on "packages" but the npm-run-all resolves operating system incompatibilities with the global scripts as well as cleans up the scripts section to be more legible.

File: package.json [hunk 1]
```diff
@@ -41,28 +42,30 @@
...
     "check-versions": "babel-node scripts/check-versions.js",
-    "format": "npm run format-packages && npm run format-cache-dir && npm run format-www && npm run format-examples && npm run format-scripts && npm run format-markdown",
+    "format": "npm-run-all -p format-packages format-cache-dir format-www format-examples format-scripts format-markdown",
     "format-cache-dir": "prettier-eslint --write \"packages/gatsby/cache-dir/*.js\"",
     "format-examples": "prettier-eslint --write \"examples/**/gatsby-node.js\" \"examples/**/gatsby-config.js\" \"examples/**/src/**/*.js\"",
     "format-packages": "prettier-eslint --write \"packages/*/src/**/*.js\"",
     "format-scripts": "prettier-eslint --write \"scripts/**/*.js\"",
     "format-markdown": "prettier --write \"**/*.md\" --semi",
     "format-www": "prettier-eslint --write \"www/*.js\" \"www/src/**/*.js\"",
+    "install-local": "yarn",
```
Comment (KyleAMathews) [inline package.json]: What's this?

Comment (danawoodman) [reply package.json]: You will get a `ERROR: Task not found: "install"` error if trying to do `npm-run-all install`

**Comment by danawoodman** [comment]
@KyleAMathews as I mentioned I don't know why Travis is failing as it gives no helpful message. Any chance you can help me troubleshoot?

**Comment by KyleAMathews** [comment]
I'm pretty sure it's failing because travisci doesn't run install first before `yarn run bootstrap` so it's not installing `npm-run-all`.

**Comment by danawoodman** [comment]
Is that something I can fix? It seems strange Travis wouldn’t run install first

**Comment by KyleAMathews** [comment]
Check our the .travisci file

**Comment by danawoodman** [comment]
@KyleAMathews just fixed the issue. Instead created a `prebootstrap` that runs `yarn` and removed the `install-local` script. Build is still failing for aformentioned test errors. Not sure how what I did could have broken that. Any ideas?

**Comment by KyleAMathews** [comment]
You should probably just add `yarn` to the `before_install` step here: https://github.com/gatsbyjs/gatsby/blob/master/.travis.yml

**Comment by danawoodman** [comment]
I don’t think that is the issue since it is just tests failing and not installl issues. Can you see the error?

**Comment by KyleAMathews** [comment]
Oh so weird! I think an upstream updated which caused some of our snapshots to break 🙄  Lemme fix it on master

**Comment by KyleAMathews** [comment]
Thanks for the PR! And persevering to the end :-)

**Comment by danawoodman** [comment]
Haha no problem. Not my first time in the rodeo 🐂🤠

**Comment by jlengstorf** [comment]
Hiya @danawoodman! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**